src/pytest_cov/plugin.py

```diff
@@ -82,6 +82,20 @@ def __call__(self, parser, namespace, values, option_string=None):
         report_type, file = values
         namespace.cov_report[report_type] = file
 
+        # coverage.py doesn't set a default file for markdown output_format
+        if report_type in ['markdown', 'markdown-append'] and file is None:
+            namespace.cov_report[report_type] = 'coverage.md'
+        if all(x in namespace.cov_report for x in ['markdown', 'markdown-append']):
+            self._validate_markdown_dest_files(namespace.cov_report, parser)
```

Comment:
It's best to move this into `__init__()` as `__call__()` is happening after `pytest` spends time on running the test and the late error reporting is rather inconvenient. It'd be nicer to tell the user that they need to fix the invocation early.